GitTfs/Core/GitTfsRemote.cs

```diff
@@ -328,7 +328,7 @@ public IFetchResult FetchWithMerge(long mergeChangesetId, bool stopOnFailMergeCo
             IEnumerable<ITfsChangeset> fetchedChangesets;
             do
             {
-                fetchedChangesets = FetchChangesets(true, lastChangesetIdToFetch);
+                fetchedChangesets = FetchChangesets(true, lastChangesetIdToFetch).ToList(); //avoid enumerating the changesets more than once
```

Comment:
I have thought a lot about this `.ToList()` and I am not sure that it should be added.
At the beginning, I thought ;)
I have added this `.ToList()` in the past but someone show me that in the `FetchChangesets()` method, the collection is constructed using a `yield return` to prevent fetching ALL the changesets and changes in these changesets in memory which could be VERY uge (and not be efficient and taking to much time) if you didn't fetch by lots. Some GigaBytes in memory which crashed the process in some cases!

To conclude, It MUST not be added to this pull request!
Please open a pull request with only this change if you think we should add this change and to open a discussion...